### PR TITLE
trim sparse branches of hierarchies when building

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

* change to use oldest kafka offset for consuming messages in case messages already exist on the environments
* refactor existng code by returning error instead of checking them and returning the error or nil.
* trim sparse branches of hierarchies when building

The solution uses multiple cypher queries as we could not figure out how to do it in a single query. We decided we could revisit this if it becomes a performance bottleneck, though it does not seem to be an expensive operation.

### How to review

review + test

### Who can review

Anyone

### CODE REVIEW

File: hierarchy/store.go [hunk 1]
```diff
@@ -229,9 +228,83 @@ func setHasData(connection bolt.Conn, instanceID, dimensionName string) error {
 	defer stmt.Close()
 
 	_, err = stmt.ExecNeo(nil)
+	return err
+}
+
+func markNodesToRemain(connection bolt.Conn, instanceID, dimensionName string) error {
```
Comment (Matt-Guest) [inline hierarchy/store.go]: Is it possible to add unit tests for some of those methods?

File: hierarchy/store.go [hunk 2]
```diff
@@ -59,6 +59,21 @@ func (store *Store) BuildHierarchy(instanceID, codeListID, dimensionName string)
 		return err
 	}
 
+	markNodesToRemain(connection, instanceID, dimensionName)
+	if err != nil {
```
Comment (eldeal) [inline hierarchy/store.go]: the errors returned from these functions aren't being captured, so currently this check is not valuable

Comment (CarlHembrough) [reply hierarchy/store.go]: Good spot! I have updated to capture errors 👍

File: hierarchy/store.go (hunk 1 above)
Comment (CarlHembrough) [reply hierarchy/store.go]: Added tests for all of the store methods 👍

**Pull request merged**